thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionJobScheduler.java

```diff
@@ -366,6 +370,44 @@ public Long runBackfill(long functionId, DateTime backfillStartTime, DateTime ba
     }
     return jobId;
   }
+  /**
+   * Different from asynchronous backfill in runBackfill, it will return after the backfill is done.
+   * @param functionId
+   * @param backfillStartTime
+   * @param backfillEndTime
+   * @param force
+   */
+  public void synchronousBackFill(long functionId, DateTime backfillStartTime, DateTime backfillEndTime, boolean force) {
+    Long jobExecutionId = runBackfill(functionId, backfillStartTime, backfillEndTime, force);
+
+    if(jobExecutionId == null) {
+      LOG.warn("Unable to perform backfill on function Id {} between {} and {}", functionId, backfillStartTime, backfillEndTime);
+      return;
+    }
+
+    JobManager jobDAO = DAO_REGISTRY.getJobDAO();
+    JobDTO jobDTO = jobDAO.findById(jobExecutionId);
+    int retryCounter = 0;
+    while(jobDTO.getStatus() != JobConstants.JobStatus.COMPLETED) {
+      if(jobDTO.getStatus() == JobConstants.JobStatus.FAILED) {
+        if(retryCounter >= MAX_BACKFILL_RETRY) {
+          LOG.warn("The backfill of anomaly function {} (task id: {}) failed. Stop retry.", functionId, jobExecutionId);
+          return;
+        }
+        retryCounter++;
+        LOG.warn("The backfill of anomaly function {} (task id: {}) failed. Retry count: {}", functionId, jobExecutionId, retryCounter);
+        jobExecutionId = runBackfill(functionId, backfillStartTime, backfillEndTime, force);
+      }
+      try {
+        TimeUnit.SECONDS.sleep(SYNC_SLEEP_SECONDS);
+      }
+      catch (InterruptedException e) {
+        LOG.warn("The backfill of anomaly function {} (task id: {}) is awakened.", functionId, jobExecutionId);
```

Comment:
The thread that monitor the backfill job is awaken, not the backfill process. You might want to refine this message.